### PR TITLE
Auto-upload to PyPi.

### PR DESCRIPTION
**Pull request by unbohn**
This implements a Github workflow that enables an auto-upload to the Python Package index (PyPi). Each time we create a new release in the Github UI, a Github Action is kicked off that will build our package and upload it to PyPI, thus eliminating the need to do this manually, and we can update ISOFIT more efficiently!

We should be ready to merge this in as soon as a respective PyPi API token is stored as secret in our repo.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (jammont) [inline .github/workflows/release.yml:11]: Should this be switched to `jammont/isofit` in preparation for the more official `isofit/isofit` image that'll be posted soon? We can wait to include that in the next PR, but something to consider.

**Review by jammont — APPROVED** [review]
Looks good, thanks for the implementation!

File: .github/workflows/release.yml
Comment (unbohn) [reply .github/workflows/release.yml:11]: Absolutely, I agree! It's just a legacy from previous times here, since I started to work on this workflow some months ago. I'd suggest that we wait until #328 is merged and I then update the image in this PR before we merge it in. I don't think we need a new PR for this.

**Review by pgbrodrick — COMMENTED** [review]
Add source commands in order to use the docker image, per @jammont.

**Pull request merged**
